zeppelin-web/src/app/search/result-list.controller.js

```diff
@@ -34,6 +36,18 @@ angular
 
       return note;
     });
+    if ($scope.notes.length === 0) {
+      $scope.isResult = false;
+    } else {
+      $scope.isResult = true;
+    }
+
+    var unbindRouteChangeEvent = $scope.$on('$routeChangeStart', function (event, next, current) {
+      if (next.originalPath !== '/search/:searchTerm') {
+        searchService.searchTerm = '';
+        unbindRouteChangeEvent();
```

Comment:
This is also not needed since the event handler is linked to the controller's scope.
The event handler is destroyed automatically